File: frontend/src/types/generalIndicatorModules.ts
export type GeneralIndicatorModule = {
  id: number;
  tagName: string;
  active: boolean;
  createdAt: string;
  updatedAt: string;
};

export type GeneralIndicatorModuleList = {
  items: GeneralIndicatorModule[];
  total: number;
  activeCount: number;
  inactiveCount: number;
};

export type GeneralIndicatorModuleSyncResult = GeneralIndicatorModuleList & {
  discoveredCount: number;
  createdCount: number;
};

File: backend/app/schemas/general_indicator_modules.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel


class GeneralIndicatorModuleResponse(BaseModel):
    id: int
    tagName: str
    active: bool
    createdAt: datetime
    updatedAt: datetime


class GeneralIndicatorModuleListResponse(BaseModel):
    items: list[GeneralIndicatorModuleResponse]
    total: int
    activeCount: int
    inactiveCount: int


class GeneralIndicatorModuleStatusRequest(BaseModel):
    active: bool


class GeneralIndicatorModuleSyncResponse(GeneralIndicatorModuleListResponse):
    discoveredCount: int
    createdCount: int
